lib/problem_check.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+class ProblemCheck
```

Comment:
This class is the hub in the new `ProblemCheck` API. Inherit it to register a new check. It provides common functionality and also acts as a registry.